src/db/sqlite.cpp

```diff
@@ -337,6 +922,77 @@ bool SqliteDbTable::put(uint32_t index, char* data, unsigned len)
     return result;
 }
 
+bool SqliteDbTable::put(Node *node)
+{
+    if (!db)
+    {
+        return false;
+    }
+
+    checkTransaction();
+
+    sqlite3_stmt *stmt;
+    bool result = false;
+
+    int sqlResult = sqlite3_prepare(db, "INSERT OR REPLACE INTO nodes (nodehandle, parenthandle, name, fingerprint, origFingerprint, type, size, share, decrypted, node) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)", -1, &stmt, NULL);
+    if (sqlResult == SQLITE_OK)
+    {
+        string nodeSerialized;
+        assert(node->serialize(&nodeSerialized));
+        sqlite3_bind_int64(stmt, 1, node->nodehandle);
+        sqlite3_bind_int64(stmt, 2, node->parenthandle);
+        std::string name = node->displayname();
+        sqlite3_bind_text(stmt, 3, name.c_str(), name.length(), SQLITE_STATIC);
+        string fp;
+        node->serializefingerprint(&fp);
+        sqlite3_bind_blob(stmt, 4, fp.data(), fp.size(), SQLITE_STATIC);
+
+        attr_map::const_iterator a = node->attrs.map.find(MAKENAMEID2('c', '0'));
+        std::string origFingerprint;
+        if (a != node->attrs.map.end())
+        {
+           origFingerprint = a->second;
+        }
+
+        sqlite3_bind_blob(stmt, 5, origFingerprint.data(), origFingerprint.size(), SQLITE_STATIC);
+
+
+        sqlite3_bind_int(stmt, 6, node->type);
+        sqlite3_bind_int64(stmt, 7, node->size);
+        int shareOrLink = 0;
+        if (node->inshare)
+        {
+            shareOrLink |= sharesOrLink_t::IN_SHARES;
+        }
+
+        if (node->outshares && node->outshares->size())
```

Comment:
In case the `user` of the `Share` pointed by the element in the map `node->outshares` is invalid, it means the folder is not shared, but a folder link. It's important to not report folder links as outshares. So we need to go through the map and check that there's at least one outshare with a valid user --> outshared. Otherwise, if there's only one item and it's shared with... no user --> folder link.

Reply:
Fixed here: https://github.com/meganz/sdk/pull/2114/files#diff-0e84897933ba9d9f28f14a3837a792b4cd4e43e692d0a372a9fcc60319ab7961R134